source/common/config/well_known_names.cc

```diff
@@ -0,0 +1,82 @@
+#include "common/config/well_known_names.h"
+
+namespace Envoy {
+namespace Config {
+
+std::unordered_map<std::string, std::string> TagNameValues::getRegexMapping() {
+  std::unordered_map<std::string, std::string> regex_mapping;
+
+  // cluster.<cluster_name>.
+  regex_mapping[CLUSTER_NAME] = "^cluster\\.((.*?)\\.)";
```

Comment:
What is the purpose of `?` here, given it follows a pattern that already matches anything? Also, `.*` is generally too powerful, since regexes will do longest match, and this will walk all over the next `.`. What you need is probably something like `^cluster\\.(([^.]*))\\.`. Same comment applies below to other regexes.

Reply:
The `?` is meant to make the quantifier non-greedy.  I'm not sure if this regex syntax is different from others, but see http://www.cplusplus.com/reference/regex/ECMAScript/ .